src/universal/mutations/DeleteProjectMutation.js

```diff
@@ -0,0 +1,47 @@
+import {commitMutation} from 'react-relay';
+import {
+  getArchiveConnection,
+  getTeamDashConnection,
+  getUserDashConnection
+} from 'universal/mutations/UpdateProjectMutation';
+import isTempId from 'universal/utils/relay/isTempId';
+import safeRemoveNodeFromConn from 'universal/utils/relay/safeRemoveNodeFromConn';
+
+const mutation = graphql`
+  mutation DeleteProjectMutation($projectId: ID!) {
+    deleteProject(projectId: $projectId) {
+      project {
+        id
+      }
+    }
+  }
+`;
+
+export const removeFromProjectConnections = (store, viewerId, projectId, teamId) => {
+  // we currently have 3 connections, user, team, and team archive
+  const viewer = store.get(viewerId);
+  const archiveConn = getArchiveConnection(viewer, teamId);
+  const teamConn = getTeamDashConnection(viewer, teamId);
+  const userConn = getUserDashConnection(viewer);
+  safeRemoveNodeFromConn(projectId, teamConn);
+  safeRemoveNodeFromConn(projectId, userConn);
+  safeRemoveNodeFromConn(projectId, archiveConn);
+};
+
+const DeleteProjectMutation = (environment, projectId, teamId, onError, onCompleted) => {
+  if (isTempId(projectId)) return undefined;
+  const {viewerId} = environment;
+  const updater = (store) => {
+    removeFromProjectConnections(store, viewerId, projectId, teamId);
```

Comment:
Rather than imperatively working with the store, can we not just use the [`NODE_DELETE`](https://facebook.github.io/relay/docs/en/mutations.html#node-delete) config for this CRUD operation?

Reply:
oh such a good question.
so those configs are a carry over from relay classic-- back then, they didn't have the imperative API or the proxy stores. you did everything with these configs. and. people. hated. it.
1. let's say Matt joins the team. do you append him or prepend him to the team members array? Neither, he is in the freakin middle
2. let's say you use a GraphQLList instead of a Connection. You have to use the imperative approach, anyways.
3. let's say you're doing something aside from the most basic add/delete, like maybe you need some arguments in your `pathToConnection`, but it's not working. How do you trace your steps?
4. IIRC RANGE_DELETE removes the node from the array, but doesn't GC the node, yet. NODE_DELETE deletes the node, and so now your array has an undefined in it, which propagates to your view layer & now you're wondering why a card is undefined... it's no fun.

FWIW, the todo-modern example uses the imperative approach while the classic todo uses the config approach, so i take that as a "yeah, we like this way better, but there's no reason to have to rewrite your old code if it's working". Personally, I don't _really_ care that much, but I'd prefer to write everything the same way instead of starting out using a config, finding it breaks or the business logic changes, then using the imperative approach anyways.